service/src/main/kotlin/no/nav/su/se/bakover/service/revurdering/RevurderingServiceImpl.kt

```diff
@@ -1219,145 +1217,112 @@ internal class RevurderingServiceImpl(
-                                }.map {
-                                    // Dersom vi skal unngå denne hacken må Iverksatt.Innvilget innholde denne istedenfor kun IDen
-                                    utbetaling = it
-                                    it.id
+        val iverksattRevurdering = when (revurdering) {
+            is RevurderingTilAttestering.IngenEndring -> {
+                log.error("Revudere til INGEN_ENDRING er ikke lov. SakId: ${revurdering.sakId}")
+                return KunneIkkeIverksetteRevurdering.IngenEndringErIkkeGyldig.left()
+            }
+            is RevurderingTilAttestering.Innvilget -> {
+                revurdering.tilIverksatt(
+                    attestant = attestant,
+                    hentOpprinneligAvkorting = { avkortingid -> avkortingsvarselRepo.hent(avkortingid) },
+                    clock = clock
+                ).map { iverksattRevurdering ->
+                    utbetalingService.genererUtbetalingsRequest(
+                        sakId = revurdering.sakId,
+                        beregning = revurdering.beregning,
+                        simulering = revurdering.simulering,
+                        attestant = attestant,
+                        uføregrunnlag = revurdering.vilkårsvurderinger.uføre.grunnlag,
+                    ).map { utbetaling ->
+                        val revurderingsfeilmelding = "Publisering av revurderingsutbetaling på køen feilet."
+                        Either.catch {
+                            sessionFactory.withTransactionContext { tx ->
+                                val vedtak = VedtakSomKanRevurderes.from(iverksattRevurdering, utbetaling.id, clock)
+                                vedtakRepo.lagre(vedtak, tx)
+                                revurderingRepo.lagre(iverksattRevurdering, tx)
+                                utbetalingService.lagreUtbetaling(utbetaling, tx)
+                                utbetalingService.publiserUtbetaling(utbetaling).mapLeft { feil ->
+                                    log.error("Kunne ikke publisere revurderingsutbetaling på køen. Ruller tilbake. SakId: ${iverksattRevurdering.sakId}", feil)
+                                    throw RuntimeException("$revurderingsfeilmelding $feil")
                                 }
-                            },
-                            hentOpprinneligAvkorting = { avkortingid ->
-                                avkortingsvarselRepo.hent(id = avkortingid)
-                            },
-                        ).map { iverksattRevurdering ->
-                            vedtak = VedtakSomKanRevurderes.from(iverksattRevurdering, utbetaling!!.id, clock).let {
-                                vedtakRepo.lagre(it)
-                                it
                             }
-                            iverksattRevurdering
+                        }.mapLeft {
+                            return if (it.message?.startsWith(revurderingsfeilmelding) == true)
```

Comment:
Litt uheldig, men ettersom `transactionContext` trenger en exception for å rulle tilbake, så mister vi typen, og må dermed gjøre denne sjekken for å kunne skille på de to feilmeldingene